components/esp_websocket_client/esp_websocket_client.c

```diff
@@ -1040,7 +1040,7 @@ static void esp_websocket_client_task(void *pv)
             int ret = esp_transport_ws_poll_connection_closed(client->transport, 1000);
             if (ret == 0) {
                 // still waiting
-                break;
+                continue;
```

Comment:
But that would loop forever if the connection closure wasn't "confirmed" by the server, correct?

We should let the client to close abruptly with "some" timeout.

But we could make the timeout configurable, e.g. in [L#1040](https://github.com/espressif/esp-protocols/pull/495/files#diff-53ead0f3896f79090b1ed1e79e1cca84ac4e505faa1bb76727ab856dabebc297L1040)